crates/sparrow-compiler/src/functions/general.rs

```diff
@@ -4,15 +4,15 @@ use crate::functions::{Implementation, Registry};
 
 pub(super) fn register(registry: &mut Registry) {
     registry
-        .register("is_valid(input: any) -> bool")
+        .register("is_valid<T: any>(input: T) -> bool")
         .with_implementation(Implementation::Instruction(InstOp::IsValid));
 
     registry
-        .register("hash(input: key) -> u64")
+        .register("hash<K: key>(input: K) -> u64")
         .with_implementation(Implementation::Instruction(InstOp::Hash));
 
     registry
-        .register("with_key(key: key, value: any, const grouping: string = null) -> any")
+        .register("with_key<K: key, T: any>(key: K, value: T, const grouping: string = null) -> T")
```

Comment:
I thought for this change we could only do 1 type parameter. This has 2

Reply:
We can do any number for this change, just not nested, like `C: Collection<K, V>`